python/hail/methods/statgen.py

```diff
@@ -201,30 +201,32 @@ def impute_sex(call, aaf_threshold=0.0, include_par=False, female_threshold=0.2,
     if aaf_threshold < 0.0 or aaf_threshold > 1.0:
         raise FatalError("Invalid argument for `aaf_threshold`. Must be in range [0, 1].")
 
-    ds = call._indices.source
-    ds, _ = ds._process_joins(call)
-    ds = ds.annotate_entries(call=call)
-    ds = require_biallelic(ds, 'impute_sex')
+    mt = call._indices.source
+    mt, _ = mt._process_joins(call)
+    mt = mt.annotate_entries(call=call)
+    mt = require_biallelic(mt, 'impute_sex')
     if (aaf is None):
-        ds = ds.annotate_rows(aaf=agg.call_stats(ds.call, ds.alleles).AF[1])
+        mt = mt.annotate_rows(aaf=agg.call_stats(mt.call, mt.alleles).AF[1])
         aaf = 'aaf'
 
+    rg = mt.locus.dtype.reference_genome
     # FIXME: filter_intervals
-    if include_par:
-        ds = ds.filter_rows(ds.locus.in_x_par() | ds.locus.in_x_nonpar())
-    else:
-        ds = ds.filter_rows(ds.locus.in_x_nonpar())
-
-    ds = ds.filter_rows(ds[aaf] > aaf_threshold)
-    ds = ds.annotate_cols(ib=agg.inbreeding(ds.call, ds[aaf]))
-    kt = ds.select_cols(
-        *ds.col_key,
-        is_female=hl.cond(ds.ib.f_stat < female_threshold,
+    mt = hl.filter_intervals(mt,
+                             hl.map(lambda x_contig: hl.parse_locus_interval(x_contig, rg), rg.x_contigs),
+                             keep=True)
+    if not include_par:
+        mt = hl.filter_intervals(mt, rg.par, keep=False)
```

Comment:
rg.par looks like a list of tuple of `(contig, start, end)`, not a list of intervals (unless the documentation is wrong). Can you make sure this branch is tested in the impute_sex tests?

Reply:
the parameter `par` is a list of tuples, but the property `par` is a list of intervals. It is documented correctly, albeit confusingly.

Reply:
confirmed that this branch is tested.

Reply:
oh, derp